services/user/Email/src/Email.cpp

```diff
@@ -0,0 +1,131 @@
+#include <psibase/dispatch.hpp>
+#include <psibase/serveContent.hpp>
+#include <psibase/serveGraphQL.hpp>
+#include <psibase/serveSimpleUI.hpp>
+#include <services/system/Accounts.hpp>
+#include <services/system/HttpServer.hpp>
+#include <services/system/commonErrors.hpp>
+#include <services/user/Email.hpp>
+#include <services/user/Events.hpp>
+#include <services/user/Nft.hpp>
+#include <services/user/REvents.hpp>
+#include <services/user/Tokens.hpp>
+
+using namespace UserService;
+using namespace psibase;
+using namespace std;
+
+namespace
+{
+   HttpRequest makeQuery(const HttpRequest& in, std::string_view s)
+   {
+      return {.host        = in.host,
+              .rootHost    = in.rootHost,
+              .method      = "POST",
+              .target      = "/sql",
+              .contentType = "application/sql",
+              .body        = std::vector(s.begin(), s.end())};
+   }
+
+   auto extractUser = [](const std::string& target) -> optional<string>
+   {
+      auto pos = target.find("?user=");
```

Comment:
This can mis-parse the URL in lots of ways. "The characters slash ("/") and question mark ("?") may represent data within the query component" ([RFC 3986](https://www.rfc-editor.org/rfc/rfc3986#page-24))

Reply:
The new logic should address this